File: Java/String1/MakeTags.java
/*
 * CodingBat Java Activity "makeTags" from String-1 section.
 * codingbat.com
 */

public class MakeTags 
{
	private static int failures = 0;
	private static final String METHOD_NAME = "makeTags";
	
	/*
	 * INSTRUCTIONS
	 * 
	 * The web is built with HTML strings like "<i>Yay</i>" which draws Yay as
	 * italic text. In this example, the "i" tag makes <i> and </i> which 
	 * surround the word "Yay". Given tag and word strings, create the HTML 
	 * string with tags around the word, e.g. "<i>Yay</i>".
	 * 
	 * makeTags("i", "Yay") -> "<i>Yay</i>"
	 * makeTags("i", "Hello") -> "<i>Hello</i>"
	 * makeTags("cite", "Yay") -> "<cite>Yay</cite>"
	 */
	public static String makeTags(String tag, String word)
	{
		//put your code ONLY here!
	}
	
	public static void main(String[] args)
	{
		testCase("\"i\"", "\"Yay\"", "\"<i>Yay</i>\"");
		testCase("\"i\"", "\"Hello\"", "\"<i>Hello</i>\"");
		testCase("\"cite\"", "\"Yay\"", "\"<cite>Yay</cite>\"");
		testCase("\"address\"", "\"here\"", "\"<address>here</address>\"");
		testCase("\"body\"", "\"Heart\"", "\"<body>Heart</body>\"");
		testCase("\"i\"", "\"i\"", "\"<i>i</i>\"");
		testCase("\"i\"", "\"\"", "\"<i></i>\"");
		
		System.out.println();
		System.out.println(failures + " failed test cases.");
	}

	private static void testCase(String a, String b, String result)
	{
		String r;
		try
		{
			r = ""+ "\"" + makeTags(a.substring(1,a.length()-1), b.substring(1,b.length()-1)) + "\"";
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.println("FAIL: ");
				failures++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.print("FAIL: ");
			failures++;
		}
		System.out.println(METHOD_NAME+"("+a+", "+b+") -> "+r + ", EXPECTED: "+result);
	}

}

File: Python/string-1/make_tags.py
'''
CodingBat Python Activity "make_tags" from string-1.
codingbat.com
'''

failures = 0

def make_tags(tag, word):
    '''
The web is built with HTML strings like "<i>Yay</i>" which draws Yay as italic text.
In this example, the "i" tag makes <i> and </i> which surround the word "Yay".
Given tag and word strings, create the HTML string with tags around the word, e.g. "<i>Yay</i>".


make_tags('i', 'Yay') → '<i>Yay</i>'
make_tags('i', 'Hello') → '<i>Hello</i>'
make_tags('cite', 'Yay') → '<cite>Yay</cite>'
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("make_tags('i', 'Yay')", '<i>Yay</i>')
    catchFailures("make_tags('i', 'Hello')", '<i>Hello</i>')
    catchFailures("make_tags('cite', 'Yay')", '<cite>Yay</cite>')
    catchFailures("make_tags('address', 'here')", '<address>here</address>')
    catchFailures("make_tags('body', 'Heart')", '<body>Heart</body>')
    catchFailures("make_tags('i', 'i')", '<i>i</i>')
    catchFailures("make_tags('i', '')", '<i></i>')

        
    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()
